src/main/java/by/onliner/listener/TestListener.java

```diff
@@ -0,0 +1,185 @@
+        try {
+            FileUtils.deleteDirectory(new File(Recorder.savePath));
+            clearFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /***
+     * When Test case get Started, this method is called.
+     */
+    @Override
+    public void onTestStart(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] STARTED");
+        try {
+            Recorder.startRecording(result.getName());
+        } catch (Exception e) {
+            logger.info("Can't start video recording");
+            e.printStackTrace();
+        }
+    }
+
+    /***
+     * When Test case get passed, this method is called.
+     */
+    @Override
+    public void onTestSuccess(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] PASSED");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /***
+     * When Test case get failed, this method is called.
+     */
+    @Override
+    public void onTestFailure(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] FAILED");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /***
+     * When Test case get Skipped, this method is called.
+     */
+    @Override
+    public void onTestSkipped(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] SKIPPED");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void onTestFailedButWithinSuccessPercentage(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] FAILED but with some percentage of success");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void onTestFailedWithTimeout(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] FAILED due to the timeout");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
```

Comment:
You should not be able to access savePath by Recorder.savePath as it breaks OOP principle (which?).
For such cases you need where and how you can store the path for the file

Reply:
Fixed